src/NUnitFramework/framework/Constraints/ExactCountConstraint.cs

```diff
@@ -31,9 +31,19 @@ namespace NUnit.Framework.Constraints
     /// item in a collection, succeeding only if a specified
     /// number of items succeed.
     /// </summary>
-    public class ExactCountConstraint : PrefixConstraint
+    public class ExactCountConstraint : Constraint
```

Comment:
There is a sense in which it already has been doing two things: apply a predicate constraint, and count the items. I modified it so that it applies a predicate constraint if one is given, and counts the items.

The benefit of a single constraint is it keeps all enumerable count-related logic together, rather than having behavior coupling across two classes which don't reference one another, and it's less confusing if you ever construct the constraints manually. The benefit of separating into two constraints isn't clear to me yet.